### ISSUE TITLE
MAINT: remove template module

### ISSUE THREAD

**Issue by bsipocz** [astropy/astroquery#2513]
We have quite a lot of very much functional and well-maintained modules atm, and I recommend following the logic (and API) in one of those when someone is proposing adding a new module. So my proposal would be to get rid of the `template_module` module. If there is anything left about the preferred API in there that is not included in the narrative docs, then move those parts to the docs.

**Comment by keflavich** [comment]
I totally agree.  The template module has not kept up with improvements to the active modules.

Let's list a set of good example modules here, then?  I can never remember... I always think that Simbad's a good one, for example, but then I look and it has a lot of custom stuff that's not applicable to all.

**Comment by bsipocz** [comment]
I would think simbad is a good one to see what methods we would like to see, in a non-complicated case, but afaik it's not systematically maintained (but indeed one that at least gets the bug fixes whenever they come up). My go-to list is alma, cadc, jplhorizons. 
And I hope the irsa one will be a good one, too once my PR lands, but anyway, let's pick one non-complicated, non-vo one and say that we keep that in top-notch shape and recommend it as the template to follow if one wants to do web scraping.

I suppose @eerovaher may have opinions here, too.

**Comment by keflavich** [comment]
The ALMA & CADC ones are great, but much more internally complicated (imo) b/c they use VO infrastructure.  They're great examples for anything VO-based, which maybe we hope will be most new modules moving forward?

**Comment by keflavich** [comment]
No, I'm wrong, the VO isn't making things more complicated, it's the infrastructure around downloading files.  Which is necessary, useful complexity, but complexity for sure.

**Comment by bsipocz** [comment]
yes, they are great as VO templates, thus my comment about choosing and picking a webscapring one and put it in top-notch shape (or just call it a day with Simbad and clean it up, but that one also uses an API, so a bit more complex than the most simple case). 

But then thinking about it, we could also pick a linelist one to show how to do line DB modules, as the "common" API methods naturally don't at all make sense for those.

**Comment by keflavich** [comment]
Oh yes, you said exactly that, I just failed to read the "non-vo" in your post.  My bad.
